mmpose/datasets/pipelines/pose3d_transform.py

```diff
@@ -0,0 +1,277 @@
+            - world_to_pixel
+            - camera_to_world
+            - camera_to_pixel
+        output_name (str|None): The name of the projected pose. If None
+            (default) is given, the projected pose will be stored in place.
+        camera_type (str): The camera class name (should be registered in
+            CAMERA).
+        camera_param (dict|None): The camera parameter dict. See the camera
+            class definition for more details. If None is given, the camera
+            parameter will be obtained during processing of each data sample
+            with the key "camara_param".
+
+    Required keys:
+        item (depend on args): Should be an array in shape [...,Kj,C]
+        camera_param (if camera parameters are not given in initialization)
+    Modified keys:
+        output_name (depend on args)
+    """
+
+    def __init__(self,
+                 item,
+                 mode,
+                 output_name=None,
+                 camera_type='SimpleCamera',
+                 camera_param=None):
+        self.item = item
+        self.mode = mode
+        self.output_name = output_name
+        self.camera_type = camera_type
+        allowed_mode = {
+            'world_to_camera',
+            'world_to_pixel',
+            'camera_to_world',
+            'camera_to_pixel',
+        }
+        if mode not in allowed_mode:
+            raise ValueError(
+                f'Got invalid mode: {mode}, allowed modes are {allowed_mode}')
+
+        if camera_param is None:
+            self.static_camera = False
+        else:
+            self.static_camera = True
+            self.camera = self._build_camera(camera_param)
+
+    def _build_camera(self, param):
+        cfgs = dict(type=self.camera_type, param=param)
+        return build_from_cfg(cfgs, CAMERAS)
+
+    def __call__(self, results):
+        assert self.item in results
+        joints = results[self.item]
+
+        if self.static_camera:
+            camera = self.camera
+        else:
+            assert 'camera_param' in results, 'Camera parameters are missing.'
+            camera = self._build_camera(results['camera_param'])
+
+        if self.mode == 'world_to_camera':
+            output = camera.world_to_camera(joints)
+        elif self.mode == 'world_to_pixel':
+            output = camera.world_to_pixel(joints)
+        elif self.mode == 'camera_to_world':
+            output = camera.camera_to_world(joints)
+        elif self.mode == 'camera_to_pixel':
+            output = camera.camera_to_pixel(joints)
+        else:
+            raise NotImplementedError
+
+        output_name = self.output_name
+        if output_name is None:
+            output_name = self.item
+
+        results[output_name] = output
+        return results
+
+
+@PIPELINES.register_module()
+class RelativeJointRandomFlip:
+    """Data augmentation with random horizontal joint flip around a root joint.
+
+    Args:
+        item (str): The name of the pose to flip.
```

Comment:
Any options?

Reply:
The options of the item value depend on the data sample from the dataset. We expect that the user should be aware of the data sample content and set the item argument properly.